lib/quoil/cli.ex

```diff
@@ -0,0 +1,61 @@
+defmodule Quoil.CLI do
+  @moduledoc """
+  Command line interface for the quoil function  
+    -> parsing the arguments
+    -> running the ping command  
+    -> interpreting the results  
+    -> logging
+    """
+
+  def main(argv) do
+    argv
+    |> Quoil.ArgsProcessor.parse_args
+    |> terminate_early?
+    |> run_ping
+    |> Quoil.ParseResults.parse_result
+    |> Quoil.LogResults.write_log
+    |> Quoil.Repeat.requested?
+
+  end
+
+
+  def terminate_early?(:help) do
+    IO.puts """
+    usage: quoil [-h | --help]
+    quoil [--interval sec] [--number nr] [--repeat nr [--wait min]] <ip_to_ping> [log_file_name]
+    
+    all values supplied must be positive integers
+
+    For more information see: https://github.com/rawdamedia/quoil
+    """
+    System.halt(0)
+  end
+  def terminate_early?(:error) do
+    IO.puts "ERROR: There was an error with the command line switches."
```

Comment:
I am only up to Prolog in Bruce Tate's "Seven Languages in Seven Weeks"... but I can't wait to get to Haskell !!!  

The blog post you linked to talks about two different libraries - Monad and MonadEx.  Just wondering:
- What benefits I get from using them on top of just passing around a Tuple where the first element is either _:ok_, _:error_ or _:help_?  
- In the example given, the error is generated by the last function in the chain, but what happens if the first function generates the error?  Does using these libraries give me some automatic bypassing of subsequent functions?  ...or will my_sum blow up if it is passed an error monad containing an error instead of a value?